website/source/docs/providers/aws/r/route.html.markdown

```diff
@@ -35,12 +35,14 @@ The following arguments are supported:
 * `destination_cidr_block` - (Required) The destination CIDR block.
 * `vpc_peering_connection_id` - (Optional) An ID of a VPC peering connection.
 * `gateway_id` - (Optional) An ID of a VPC internet gateway or a virtual private gateway.
```

Comment:
Hi - yes, the fields in the input struct are different:

https://github.com/hashicorp/terraform/pull/4381/files#diff-a9b45e521c3ff848176f0c6492abadabR119 (`GatewayId`) vs https://github.com/hashicorp/terraform/pull/4381/files#diff-a9b45e521c3ff848176f0c6492abadabR125 (`NatGatewayId`)

I've not looked at the Console but I'd imagine it aggregates them and then chooses the correct option prior to making the API call.